src/fairplay.js

```diff
@@ -0,0 +1,40 @@
+import { stringToArray } from './util';
+
+export default function concatInitDataIdAndCertificate(initData, id, certificate) {
+  if (typeof id === 'string') {
```

Comment:
`id` is `contentId` which is based off of `initData`, in the previous function it converts the `event.initData`, a `Uint8Array`, into a string, into a anchor tag, to pull off the `hostname`, which leaves you at a string.

this function needs a bunch of `Uint8Array`'s and `contentId`, or `id` as seen above, is the only one that can be a string—needs to be converted back into a `Uint8Array`.

as to when it would never be a string? not sure,